zerver/tests/test_messages.py

```diff
@@ -1062,6 +1075,20 @@ def test_message_fail_to_announce(self) -> None:
                                                      "subject": "Test subject"})
         self.assert_json_error(result, "Only organization administrators can send to this stream.")
 
+        # Notification bot is no longer a cross-realm bot.
+        with self.settings(CROSS_REALM_BOT_EMAILS=["not-notification-bot@zulip.com"]):
+            notification_bot = get_system_bot("notification-bot@zulip.com")
+            password = "test_password"
+            notification_bot.set_password(password)
+            notification_bot.save()
+            self.login(notification_bot.email, password)
```

Comment:
I don't like using `self.login` for bot users -- they can't login.

We should use the API for this instead.

That said, I'm not convinced we need this test at all; might be better to just remove it.

Reply:
@timabbott I've removed the test.